visualize-admin/visualization-tool
# perf: Initial DataCubeComponents / dimension values query optimizations

Closes #1372.

This PR is a first step towards optimizing the performance situation related to `DataCubeComponents` query.

- Disabled loading of dimension values where applicable
- Adjusted several places in the code to not send unnecessary filters to the server to make sure we use `sh:in` as much as possible
- Fixed an issue with not using optional chaining to access last dimension value
- Refactored the approach to fetching available filter values from one combined `DataCubesComponents` call with filters for every dimension to individual calls directly from filter components. This enables us to more precisely pass filters and rely more on `sh:in`, as for the first filter we won't send `filters` object anymore
- Fixed an issue with fetching dimension values which resulted in fetching values that couldn't be selected (combinations resulting in no observations, fixed after selection by possible filters query)

### Next steps
- Consolidate `DataCubeComponents`-related queries into `CONSTRUCT` query/ies
- Consider using `mobx` to improve manageability of data fetching across the app
- Consider decoupling dimension values fetching from components to make it more clear where we actually need to fetch values (currently it's not ideal, as we might or might not have values fetched)

### How to test
**PR**
1. Go to [this link](https://visualization-tool-git-perf-optimize-data-cube-comp-527a80-ixt1.vercel.app/en/create/new?cube=https://environment.ld.admin.ch/foen/fab_ahst_exp_imp_pipeline/1&dataSource=Test).
2. ✅ See that the filters are loaded in the background (some might fail to load due to Vercel's 15s timeout).
3. ✅ See that the last filter (`4. Zusatzmengencode - EN`) contains five values.

**TEST**
1. Go to [this link](https://test.visualize.admin.ch/en/create/new?cube=https://environment.ld.admin.ch/foen/fab_ahst_exp_imp_pipeline/1&dataSource=Test).
2. ❌ See that the initial view loads much slower compared to the PR link.
3. ❌ See that the last filter (4. Zusatzmengencode - EN) contains seven values.
4. Try to change value of above filter to `Metres`.
5. ❌ See that the chart reloads back to `No supplementary unit`.
